appeals/web/src/server/appeals/appeal-details/lpa-questionaire/outcome-incomplete/outcome-incomplete.controller.js

```diff
@@ -0,0 +1,289 @@
+import logger from '../../../../lib/logger.js';
+import { mapIncompleteReasonsToCheckboxItemParameters } from '../lpa-questionnaire.mapper.js';
+import { getLPAQuestionnaireIncompleteReasons } from '../lpa-questionnaire.service.js';
+import { objectContainsAllKeys } from '../../../../lib/object-utilities.js';
+import { getIdByNameFromIdNamePairs } from '../../../../lib/id-name-pairs.js';
+import { webDateToDisplayDate } from '../../../../lib/dates.js';
+
+/**
+ *
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ */
+const renderIncompleteReason = async (request, response) => {
+	const { errors, body } = request;
+
+	if (
+		!objectContainsAllKeys(request.session, ['appealId', 'appealReference', 'lpaQuestionnaireId'])
+	) {
+		return response.render('app/500.njk');
+	}
+
+	const {
+		// @ts-ignore
+		appealId,
+		// @ts-ignore
+		appealReference,
+		// @ts-ignore
+		lpaQuestionnaireId
+	} = request.session;
+
+	// @ts-ignore
+	const existingWebLPAQuestionnaireReviewOutcome = request.session.webLPAQuestionnaireReviewOutcome;
+
+	if (
+		existingWebLPAQuestionnaireReviewOutcome &&
+		(existingWebLPAQuestionnaireReviewOutcome.appealId !== appealId ||
+			existingWebLPAQuestionnaireReviewOutcome.lpaQuestionnaireId !== lpaQuestionnaireId ||
+			existingWebLPAQuestionnaireReviewOutcome.validationOutcome !== 'incomplete')
+	) {
+		// @ts-ignore
+		delete request.session.webLPAQuestionnaireReviewOutcome;
+	}
+
+	// @ts-ignore
+	const { webLPAQuestionnaireReviewOutcome } = request.session;
```

Comment:
No, it's not required as it is an optional var in case the session already exists and, render incomplete page and `body['review-outcome'] doesn't exist